src/Test/Legacy/WebServerListener.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\Test\Legacy;
+
+use FOS\HttpCache\Test\WebServerListenerTrait;
+use PHPUnit\Framework\TestListener;
+
+/**
+ * A PHPUnit test listener that starts and stops the PHP built-in web server.
+ *
+ * This listener is configured with a couple of constants from the phpunit.xml
+ * file. To define constants in the phpunit file, use this syntax:
+ * <php>
+ *     <const name="WEB_SERVER_HOSTNAME" value="localhost" />
+ * </php>
+ *
+ * WEB_SERVER_HOSTNAME host name of the web server (required)
+ * WEB_SERVER_PORT     port to listen on (required)
+ * WEB_SERVER_DOCROOT  path to the document root for the server (required)
+ */
+class WebServerListener implements TestListener
```

Comment:
if i understand my random experiments correctly, the class_alias thing merges the `use` statements from both files. to work around, i suggest you alias the TestListener in this file to LegacyTestListener.